### ISSUE TITLE
Add podman support

### ISSUE THREAD

**Issue by goldmann** [cekit/cekit#363]
We need to think if we can make it the default. Or we can default to what;s available on the machine, by priority.

**Comment by vorburger** [comment]
as per #380, wouldn't be possible to specify only that you just need _a_ `docker` CLI (from whatever package provides it), instead of being bound to and require a specific package like `docker-common` (as you currently have it) or `podman-docker` (as you are planning here) ?

**Comment by rnc** [comment]
@vorburger I believe the current docker support talks directly to the daemon as opposed to the buildah support which initiates the program directly. For podman support - it could use the python varlink interface or potentially invoke the CLI as well.

**Comment by goldmann** [comment]
@vorburger Cekit 3.0 won't require a specific package to make it work with Docker. More information can be found in my comment: https://github.com/cekit/cekit/issues/380#issuecomment-463131387. Cekit 2.2 dependencies won't be changed because the only thing that will go into 2.2 are bugfixes.

@rnc is correct, we use the python library to interact with Docker. We used to call the CLI directly, but we moved away from it. It's unclear how we will provide support for Podman yet since work on this ticket hasn't been started.
